docs/architecture/adr-061-core-module-api.md

```diff
@@ -0,0 +1,490 @@
+# ADR 061: Core Module API
+
+## Changelog
+
+* 2022-08-18 First Draft
+
+## Status
+
+PROPOSED Not Implemented
+
+## Abstract
+
+A new core API is proposed to replace the existing `AppModule` and `sdk.Context` frameworks a set of core services
+and a new `Handler` struct. This core API aims to:
+- be simpler,
+- more extensible
+- and more stable than the current framework
+while also
+- enabling deterministic events and queries, and
+- supporting event listeners
+- and [ADR 033: Protobuf-based Inter-Module Communication](./adr-033-protobuf-inter-module-comm.md) clients.
+
+## Context
+
+Historically modules have exposed their functionality to the state machine via the `AppModule` and `AppModuleBasic`
+interfaces which have the following shortcomings:
```

Comment:
I don't know what this means? I recon you could argue they expose their functionality via Keepers. The `AppModule*` interfaces are really more app wiring UX primitives.

Reply:
Maybe "state machine" isn't the right term. They expose their functionality to the SDK framework/BaseApp with these interfaces